### PR TITLE
use ensure_length instead of length for memory allocation

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Connects to https://github.com/ros2/rosidl/issues/114

### CODE REVIEW

File: rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template [hunk 1]
```diff
@@ -182,7 +182,7 @@ convert_ros_to_dds(const void * untyped_ros_message, void * untyped_dds_message)
       return false;
     }
     DDS_Long length = static_cast<DDS_Long>(size);
-    dds_message->@(field.name)_.length(length);
+    dds_message->@(field.name)_.ensure_length(length, 2 * length);
```
Comment (dirk-thomas) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: Why allocate twice as much memory then needed?

Comment (wjwwood) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: That's the max, it only allocates if there is not enough memory of length. I suppose, instead, it should be something like `current_capacity * 2`.

Comment (wjwwood) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: If the `dds_message` is temporary, then as @dirk-thomas suggested it should be `length, length`. @mikaelarguedas can you check to make sure this message is temporary and update if it is?

Comment (dirk-thomas) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: I am not sure if the `dds_message` instance exists longer then for a single serialization but if not I think `(length, length)` would be better to not allocate more memory then needed.

Comment (mikaelarguedas) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: Yes that's my feeling too, the message should be temporary, didn't have time to look more into it but will give an update soon

Comment (mikaelarguedas) [inline rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template]: Message is indeed temporary. Setting the allocated memory to `length`

**Comment by jacquelinekay** [comment]
There's similar calls in `rosidl_typesupport_connext_cpp`, does it need to be changed there as well?

https://github.com/ros2/rmw_connext/blob/master/rosidl_typesupport_connext_cpp/resource/msg__type_support.cpp.template#L116

**Comment by mikaelarguedas** [comment]
Yeah it would make sense in the case that `is_upper_bound` is set and that the current `maximum` is less than the upper bound. Then I don't know if this scenario can actually happen.
In the end I would say it's safer to change it anyway.

**Comment by dirk-thomas** [comment]
+1

**Comment by wjwwood** [comment]
Can this just be merged?

**Pull request merged**